ProcessLib/ProcessVariable.cpp

```diff
@@ -202,11 +219,99 @@ ProcessVariable::createBoundaryConditions(
         bcs.push_back(std::move(bc));
     }
 
+    if (_deactivated_subdomains.empty())
+        return bcs;
+
+    createBoundaryConditionsForDeactivatedSubDomains(dof_table, variable_id,
+                                                     parameters, bcs);
     return bcs;
 }
 
-std::vector<std::unique_ptr<SourceTerm>>
-ProcessVariable::createSourceTerms(
+void ProcessVariable::createBoundaryConditionsForDeactivatedSubDomains(
+    const NumLib::LocalToGlobalIndexMap& dof_table, const int variable_id,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    std::vector<std::unique_ptr<BoundaryCondition>>& bcs)
+{
+    auto& parameter = findParameter<double>(
+        DeactivatedSubdomain::name_of_paramater_of_zero, parameters, 1);
+
+    for (auto const& deactivated_subdomain : _deactivated_subdomains)
+    {
+        auto const& deactivated_sudomain_meshes =
+            (*deactivated_subdomain).deactivated_sudomain_meshes;
+        for (auto const& deactivated_sudomain_mesh :
+             deactivated_sudomain_meshes)
+        {
+            for (int component_id = 0;
+                 component_id < dof_table.getNumberOfComponents();
+                 component_id++)
+            {
+                // Copy the time interval.
+                std::unique_ptr<BaseLib::TimeInterval> time_interval =
+                    std::make_unique<BaseLib::TimeInterval>(
+                        (*(*deactivated_subdomain).time_interval));
```

Comment:
I'd think that `(*(*deactivated_subdomain).time_interval)` is equivalent to `*deactivated_subdomain->time_interval`.

Reply:
Changed.

Reply:
Changed the remaining stuffs by amending commit b4200a7.